build/nvcr.Dockerfile

```diff
@@ -0,0 +1,98 @@
+# Copyright The FMS HF Tuning Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+## Global Args #################################################################
+## If the nvcr container is updated, ensure to check the torch and python
+## installation version inside the dockerfile before pushing changes.
+ARG NVCR_IMAGE_VERSION=25.02-py3
+
+# This is based on what is inside the NVCR image already
+ARG PYTHON_VERSION=3.12
+
+## Base Layer ##################################################################
+FROM nvcr.io/nvidia/pytorch:${NVCR_IMAGE_VERSION} AS dev
+
+ARG USER=root
+ARG USER_UID=0
+ARG WORKDIR=/app
+ARG SOURCE_DIR=/app/fms-hf-tuning
+ARG SOURCE_BRANCH=main
+
+ARG ENABLE_FMS_ACCELERATION=true
+ARG ENABLE_AIM=true
+ARG ENABLE_ALORA=true
+ARG ENABLE_MLFLOW=true
+ARG ENABLE_SCANNER=true
+ARG ENABLE_CLEARML=true
+ARG ENABLE_TRITON_KERNELS=true
+ARG ENABLE_MAMBA_SUPPORT=false
+
+# Ensures to always build mamba_ssm from source
+ENV PIP_NO_BINARY=mamba-ssm,mamba_ssm
+
+RUN python -m pip install --upgrade pip
+
+# upgrade torch as the base layer contains only torch 2.7
+RUN pip install --upgrade --force-reinstall torch torchaudio torchvision --index-url https://download.pytorch.org/whl/cu128
+
+# Install main package + flash attention
+RUN git clone --branch ${SOURCE_BRANCH} --depth 1 https://github.com/foundation-model-stack/fms-hf-tuning.git ${SOURCE_DIR}
```

Comment:
It should not be using git clone, it should be copy from local, since it is from a local build.